src/audio/mixer/mixer.c

```diff
@@ -286,7 +165,10 @@ static int mixer_reset(struct comp_dev *dev)
 			struct comp_buffer *source = container_of(blist, struct comp_buffer,
 								  sink_list);
 			struct comp_buffer __sparse_cache *source_c = buffer_acquire(source);
-			bool stop = mixer_stop_reset(dev, source_c->source);
+			bool stop = false;
+
+			if (source_c->source && source_c->source->state > COMP_STATE_READY)
+				stop = true;
```

Comment:
`stop = source_c->source && source_c->source->state > COMP_STATE_READY`

Reply:
@lyakh no - I can read the former without having to think too hard.